core/src/test/java/greencity/controller/HabitStatisticControllerTest.java

```diff
@@ -0,0 +1,166 @@
+package greencity.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import greencity.ModelUtils;
+import greencity.dto.habitstatistic.*;
+import greencity.service.HabitStatisticService;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+class HabitStatisticControllerTest {
+    private MockMvc mockMvc;
+    @Mock
+    HabitStatisticService habitStatisticService;
+    @InjectMocks
+    HabitStatisticController habitStatisticController;
+
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    long existId = 1;
+
+    @BeforeEach
+    void setUp() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(habitStatisticController)
+                .build();
+    }
+
+    @Test
+    void findAllByHabitId() throws Exception {
+        GetHabitStatisticDto habitStatisticDto = new GetHabitStatisticDto();
+        habitStatisticDto.setHabitStatisticDtoList(Collections.singletonList(new HabitStatisticDto()));
+
+        when(habitStatisticService.findAllStatsByHabitId(1L)).thenReturn(habitStatisticDto);
+        mockMvc.perform(get("/habit/statistic/{habitId}", existId));
+
+        ResponseEntity<GetHabitStatisticDto> response = habitStatisticController.findAllByHabitId(existId);
+
+        Assertions.assertEquals(HttpStatus.OK, response.getStatusCode());
+        Assertions.assertEquals(habitStatisticDto, response.getBody());
+    }
+
+    @Test
+    void findAllStatsByHabitAssignId() throws Exception {
+        List<HabitStatisticDto> habitStatisticDtoList = Collections.singletonList(new HabitStatisticDto());
+
+        when(habitStatisticService.findAllStatsByHabitAssignId(existId)).thenReturn(habitStatisticDtoList);
+        mockMvc.perform(get("/assign/{habitAssignId}", existId));
```

Comment:
You forgot about "/habit/statistic" request mapping. So now mockMvc returns 404 (not found).
And also use .andExpect(status.isOk()).

Reply:
Also you can extract base link part like private static final String habitsStatisticsLink = "/habit/statistic";

Reply:
thank you for the comments, I have made fixes